libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -259,11 +259,28 @@ string IRGenerator::constructorCode(ContractDefinition const& _contract)
 
 	if (constructor)
 	{
-		solUnimplementedAssert(constructor->parameters().empty(), "");
+		ABIFunctions abiFunctions(m_evmVersion, m_context.revertStrings(), m_context.functionCollector());
+		unsigned paramVars = make_shared<TupleType>(constructor->functionType(false)->parameterTypes())->sizeOnStack();
+
+		Whiskers t(R"X(
+			let dataSize := datasize("<object>")
+			let argSize := sub(codesize(), dataSize)
```

Comment:
`codesize()` should contain program data plus parameters, so size of parameters should be diff of those two